core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -824,10 +824,11 @@ protected void convertOrder(
       List<SqlNode> orderExprList,
       SqlNode offset,
       SqlNode fetch) {
-    if (!bb.top
+    boolean keepSort = config.isKeepSortInSubQuery() || bb.top;
+    if (!keepSort
```

Comment:
Can we abstract the common logic `config.isKeepSortInSubQuery() || bb.top` into a method there ?
```java
/**
* Returns whether we should keep the sort for the subsequent query conversion. 
*
* @param topQuery Whether the query is in the top level.
*/
boolean keepSortInSubQuery(boolean topQuery) {
  return config.isKeepSortInSubQuery() || topQuery;
}
```

Reply:
> Can we abstract the common logic `config.isKeepSortInSubQuery() || bb.top` into a method there ?
> 
> ```java
> /**
> * Returns whether we should keep the sort for the subsequent query conversion. 
> *
> * @param topQuery Whether the query is in the top level.
> */
> boolean keepSortInSubQuery(boolean topQuery) {
>   return config.isKeepSortInSubQuery() || topQuery;
> }
> ```

Thanks, nice advice, the code is more readable.